store/sqlstore/upgrade.go

```diff
@@ -803,3 +808,81 @@ func upgradeDatabaseToVersion524(sqlStore SqlStore) {
 		saveSchemaVersion(sqlStore, VERSION_5_24_0)
 	}
 }
+
+func upgradeDatabaseToVersion525(sqlStore SqlStore) {
+	//if shouldPerformUpgrade(sqlStore, VERSION_5_24_0, VERSION_5_25_0) {
+	saveSchemaVersion(sqlStore, VERSION_5_25_0)
+	//}
+}
+
+func upgradeDatabaseToVersion526(sqlStore SqlStore) {
+	//if shouldPerformUpgrade(sqlStore, VERSION_5_25_0, VERSION_5_26_0) {
+	err := precheckMigrationToVersion526(sqlStore)
```

Comment:
Isn't this specific to the changes below? I am wondering whether it will still be applicable if more changes are added here?

We are considering this entire function to be atomic, but will that be always the case? Otherwise it might make sense to only skip adding the 4 column changes rather than returning.

Reply:
Yes that function is performing specific checks for the changes below. If more changes are included And those should be checked before executing then we should add more checks in that function.

We usually don’t need to do this part but I’m the AlterColumn case if one of the fails we leave the database screens inconsistent.

In my opinion the migration should be all or nothing so if one of the checks fails e we should abort the entire migration

Reply:
Sounds good to me.